### DISCUSSION TITLE
Post automatically every hour

### DISCUSSION THREAD

**Discussion by fotoente** [yupix/Mi.py#86] [Q&A]
Hello,

I am trying to use the framework to make the bot posting a note every hour.

I can get the loop starting and it works like a charm with

```
@tasks.loop(3600)
    async def loop1h(self):
        #res = await self.post_note(cfpost()+" Autopost! 1 Hour")
        #print(res.content)
        print (datetime.now().strftime('%Y-%m-%d %H:%M:%S')+" Hour Loop")
```
and `self.loop1h.start()  #Launching 1 hour task` in `on_ready()`

But as soon I use `await self.post_note()` it doens't work anymore. I don't get any error message or something like that. It just not working anymore. The time date & date doesn't show up either. (Additional information `cfpost()` just returns a String, but even without that function it doesn't do anything)

I really can't find the error...

**Comment by yupix** [comment]
It will take some time to investigate this. We apologize for any inconvenience this may cause.

**Reply by yupix** [reply]
The detailed cause of this problem has not been determined, but this problem can be solved by using a variable that instantiates a class that inherits from commands. I'll let you know if I find out anything else.

```python
class MyBot(commands.Bot):
    def __init__(self):
        super().__init__()

    @tasks.loop(15)
    async def test_loop(self):
        print('working loop')
        await bot.post_note('test') # change self to bot

    async def on_ready(self, ws):
        print(f'Connecting {self.i.username}')
        await Router(ws).connect_channel(['global', 'main'])
        self.test_loop.start()


if __name__ == '__main__':
    url = ''
    token = ''
    bot = MyBot()  # this
    asyncio.run(bot.start(url, token))
  ```

**Reply by yupix** [reply]
I found the cause, but the cause is that the function received by the loop decorator does not have enough self, so I'm trying to figure out what to do.

**Reply by yupix** [reply]
You can see the progress of this issue at:
https://argon.teamblackcrystal.com/issue/MP-24

**Reply by yupix** [reply]
I was able to fix the problem, but by the way, I'm making destructive changes to make changes to arguments, etc., so please wait until v3.0.0 is released.
